keps/sig-storage/1790-recover-resize-failure/README.md

```diff
@@ -288,73 +287,70 @@ after expansion is complete even with older kubelets. No recovery from expansion
+        - Components exposing the metric: kube-controller-manager
+    - node expansion operation duration:
+        - Metric name: storage_operation_duration_seconds{operation_name=volume_fs_resize}
+        - [Optional] Aggregation method: percentile
+        - Components exposing the metric: kubelet
+    - node expansion operation errors:
+        - Metric name: storage_operation_errors_total{operation_name=volume_fs_resize}
+        - [Optional] Aggregation method: cumulative counter
+        - Components exposing the metric: kubelet
   - [ ] Other (treat as last resort)
     - Details:
 
 * **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
-  At the high-level this usually will be in the form of "high percentile of SLI
-  per day <= X". It's impossible to provide a comprehensive guidance, but at the very
-  high level (they needs more precise definitions) those may be things like:
-  - per-day percentage of API calls finishing with 5XX errors <= 1%
-  - 99% percentile over day of absolute value from (job creation time minus expected
-    job creation time) for cron job <= 10%
-  - 99,9% of /health requests per day finish with 200 code
+  After this feature is rolled out, there should not be any increase in 95-99 percentile of
+  both `expand_volume` and `volume_fs_resize` durations. Also the error rate should not increase for 
+  `storage_operation_errors_total` metric.
 
 * **Are there any missing metrics that would be useful to have to improve
   observability if this feature?**
-  Describe the metrics themselves and the reason they weren't added (e.g. cost,
-  implementation difficulties, etc.).
-
-### Dependencies
+  We are planning to add new counter metrics that will record success and failure of recovery operations.
+  In cases where recovery fails, the counter will forever be increasing until an admin action resolves the error.
 
-_This section must be completed when targeting beta graduation to a release._
-
-* **Does this feature depend on any specific services running in the cluster?**
-  Think about both cluster-level services (e.g. metrics-server) as well
-  as node-level agents (e.g. specific version of CRI). Focus on external or
-  optional services that are needed. For example, if this feature depends on
-  a cloud provider API, or upon an external software-defined storage or network
-  control plane.
+  Tentative name of metric is - `operation_operation_volume_recovery_total{state='success', volume_name='pvc-abce'}`
 
-  For each of the fill in the following, thinking both about running user workloads
-  and creating new ones, as well as about cluster-level services (e.g. DNS):
-  - [Dependency name]
-    - Usage description:
-      - Impact of its outage on the feature:
-      - Impact of its degraded performance or high error rates on the feature:
+  The reason of using PV name as a label is - we do not expect this feature to be used in a cluster very often
```

Comment:
Please get an opinion from sig-instrumentation on your PR.  I think even knowing the overall success and fail counts is useful.

Reply:
cc @dgrisonnet . What are you opinions on emitting a metric that can potentially contain volume name as a label? The motivation of using volume name here is - this metric should be emitted relatively rarely (not everyday people are expected to need to recover from volume expansion failures) and hence use of volume name should be okay.

Reply:
I agree with David, knowing the overall success and failures is definitely useful, however, I am wondering if we really need the volume name here. I don't have much knowledge of storage so could you perhaps walk me through a scenario where volume expansion failures are only happening for one specific volume and how a cluster administrator could mitigate this issue?

Reply:
1. The user could have made a typo while editing PVC. Say you have 10GB PVC and you want to expand to 100GB, but instead type `1000GB`. Now expansion is stuck forever because there may not be enough space in backend to satisfy 1000GB and hence this feature allows users to retry expansion with a lower value (say 100GB). 
2. May be not a typo but available capacity was not obvious to the user and they increased the size to a value which can't be fulfilled. 

So those are the scenarios. It still should not be that high.

Reply:
Awesome thank you for the insight. Based on the scenarios you mentioned, I think it should be fine to add a `volume` label to the counter metrics.